File: Implementierung/Matlab/JensImplementierung_altered/compute_K_from_a.m
function [ K ] = compute_K_from_a( a, verbosity )

    b=a;
    du=1;
    
    M_=round(300/du);
    M=M_*2+1;
    K=zeros(M,2);
    L=length(b);

    for i=1:M
        K(i,1)=-M_+(i-1)*du;
        for ind=1:L
            K(i,2)=K(i,2)+b(ind)*K(i,1)^(ind);
        end
    end
    
    if verbosity
        figure;
        plot(K(:,1),K(:,2));
        title('Kennlinie')
        xlabel('U_in in mV')
        ylabel('U_out in mV')
    end

end

File: Implementierung/Python/nichtLinear/compute/compute_K_from_a.py
# -*- coding: utf-8 -*-
"""
Created on Wed Aug 23 12:09:14 2017

@author: denys
"""
def compute(a, verbosity):
        
    import matplotlib.pyplot as plt
    import numpy as np
    from helpers import csvHelper



    b = a;
    du = 1;

    M1=round(300/du)
    M=int(M1*2+1)
    K=np.zeros((M,2))
    L=len(b) 
    
    for i in range (0,M):
        K[i,0]=-M1+(i)*du
        for ind in range (0,L):
            K[i,1]=K[i,1]+b[ind]*K[i,0]**(ind+1)
    

    if verbosity:
        plt.figure
        plt.plot(K[:,0],K[:,1])
        plt.title('Kennlinie')
        plt.xlabel('U_in in mV')
        plt.ylabel('U_out in mV')
        plt.show()
    
    return(K)



